src/ast/nodes/shared/FunctionNode.ts

```diff
@@ -45,6 +46,14 @@ export default class FunctionNode extends FunctionBase {
 
 	hasEffects(context: HasEffectsContext): boolean {
 		if (!this.deoptimized) this.applyDeoptimizations();
+
+		if (
+			(this.context.options.treeshake as NormalizedTreeshakingOptions).annotations &&
+			this.annotations
```

Comment:
If we use a different annotation, of course it means we cannot use this check as there are different kinds of annotations with different effects.
Here would be one way of handling it:
- we still create `this.annotations` containing ALL annotations because we need this to remove all annotations of a Node if the Node is tree-shaken (it contains the exact locations of the comments we need to remove)
- These annotations are collected in `pureComments.ts`. To avoid re-reading all annotation comments, we add additional flags. Currently it works by adding a `_rollupAnnotations` property to the acorn Node, which is translated in `Node.ts` to the `annotations` property. How about depending on the annotation, we additionally add a `_rollupPure` or `_rollupNoSideEffects` property to the acorn Node that is copied over to the Rollup Nodes?
- We just need to make sure we also adapt the code in `CallExpression.ts` and `NewExpression.ts`.